### PR TITLE
Custom Element support

### LINKED ISSUE #110: createDom produces intermediate DOM states
I am trying to combine Maquette and A-Frame to render a WebVR scene and am running into an issue with intermediate DOM states. 

When Maquette creates a new DOM node it attaches the new DOM node to the parent node before adding attributes and properties: https://github.com/AFASSoftware/maquette/blob/master/src/maquette.ts#L789-L795. This causes an issue with A-Frame which uses Custom Elements that watch for and react to changes in the DOM. When Maquette attaches the new DOM node to the parent, it is incomplete and thus creates an intermediate and invalid DOM.

These intermediate DOM states are problematic because they're not rendering what the developer expects and they prematurely trigger observers and proxies.

As a comparison, A-Frame resolves these issues w/ React by explicitly updating attributes in `componentDidUpdate()`: https://github.com/aframevr/aframe-react/blob/master/src/index.js#L91. But I don't think Maquette offers a similar mechanism to update the DOM before it's attached. 

I think the most straight-forward solution would be to call `initPropertiesAndChildren()` on `domNode` right before it's added to the DOM.

### PR DESCRIPTION
**Pull request by devpaul**
Added support for custom elements by delaying their attachment to the DOM until after attributes and properties have been attached.

Resolves #110

### CODE REVIEW

**Comment by devpaul** [comment]
Something isn't working right.

File: src/maquette.ts [hunk 1]
```diff
@@ -749,17 +749,29 @@ let initPropertiesAndChildren = function(domNode: Node, vnode: VNode, projection
...
     domNode = vnode.domNode = doc.createTextNode(vnode.text!);
-    if (insertBefore !== undefined) {
-      parentNode.insertBefore(domNode, insertBefore);
-    } else {
-      parentNode.appendChild(domNode);
-    }
+    insertNode(parentNode, domNode, vnode, insertBefore);
   } else {
     for (i = 0; i <= vnodeSelector.length; ++i) {
       c = vnodeSelector.charAt(i);
```
Comment (johan-gorter) [inline src/maquette.ts]: Rather than using a regex, testing if c === '-' would be more performant

**Comment by johan-gorter** [comment]
I think that using a regex would be too much of a performance penalty. The selector is already parsed character by character, this would be a better place to test for '-'

**Comment by johan-gorter** [comment]
@devpaul we are currently considering if we couldn't better delay attaching all nodes in a new major/breaking release of maquette. This would force non-custom-component plugins to register a callback to delay initialisation until the projector is finished building the DOM. This would also mean you can prevent unneeded reflows from happening.

**Comment by devpaul** [comment]
@johan-gorter that would be great! My two biggest pain points in working with custom elements is failed initialization due to missing attributes and unnecessary reflows. I imagine most custom elements are designed and tested to work as a HTML component, but not necessarily to work in a vdom. Having the ability to help my framework work around these corner cases would mean a great deal.

Please let me know if I can help or provide feedback.

**Comment by RickHoving** [comment]
@devpaul we are adressing this issue in [maquette 3.0](https://dev.afas.nl/blog-dev/ideas-for-maquette-3) and would like your oppinion on this subject. See #118

**Pull request closed without merging**